### PR TITLE
Enable BreakingChangeAnalysis only when running in Azure

### LINKED ISSUE #6253: Automatically Opt Out BlobChangeAnalysisStateProvider When Hosting Environment is not App Service
The function host logs the below exception while deploying to K8S without a Azure Storage Account. The exception does not break the host but just noisy, and can be misleading for customers who run function apps not in app service.

```
System.InvalidOperationException: The BlobChangeAnalysisStateProvider requires the default storage account 'Storage', which is not defined.
   at Microsoft.Azure.WebJobs.Script.ChangeAnalysis.BlobChangeAnalysisStateProvider.GetCurrentAsync(CancellationToken cancellationToken) in /src/azure-functions-host/src/WebJobs.Script.WebHost/BreakingChangeAnalysis/BlobChangeAnalysisStateProvider.cs:line 40
   at Microsoft.Azure.WebJobs.Script.ChangeAnalysis.ChangeAnalysisService.TryLogBreakingChangeReportAsync(CancellationToken cancellationToken) in /src/azure-functions-host/src/WebJobs.Script.WebHost/BreakingChangeAnalysis/ChangeAnalysisService.cs:line 92
```

[Proposed Solution]
The function host can sniff out hosting environment and decide to opt out the BlobChangeAnalysisStateProvider feature if it is not supposed to run in the given hosting environment.

### PR DESCRIPTION
**Pull request by pragnagopa**
Fixes #6253

### CODE REVIEW

**Comment by pragnagopa** [comment]
@fabiocav - gentle ping for review as the sprint is closing today.

**Comment by pragnagopa** [comment]
ping for review!

**Review by fabiocav — DISMISSED** [review]

File: src/WebJobs.Script.WebHost/DependencyInjection/DependencyValidator/DependencyValidator.cs [hunk 1]
```diff
@@ -48,7 +48,6 @@ private static ExpectedDependencyBuilder CreateExpectedDependencies()
                 .Expect<FileMonitoringService>()
                 .Expect<WorkerConsoleLogService>()
                 .Expect<FunctionInvocationDispatcherShutdownManager>()
-                .Expect<ChangeAnalysisService>()
```
Comment (pragnagopa) [inline src/WebJobs.Script.WebHost/DependencyInjection/DependencyValidator/DependencyValidator.cs:51]: @fabiocav - Removed this for tests. Let me know if I should add unit tests for checking this list based on environment

Comment (fabiocav) [reply src/WebJobs.Script.WebHost/DependencyInjection/DependencyValidator/DependencyValidator.cs:51]: You can flag this as optional

Comment (pragnagopa) [reply src/WebJobs.Script.WebHost/DependencyInjection/DependencyValidator/DependencyValidator.cs:51]: done! thanks

**Review by fabiocav — APPROVED** [review]

**Pull request merged**